File: reorder-ids.js
const fs = require('fs');

// Baca file data.json
const data = JSON.parse(fs.readFileSync('./data.json', 'utf-8'));

// Urutkan ulang ID dari 1 sampai jumlah total soal
data.forEach((question, index) => {
  question.id = index + 1;
});

// Tulis kembali ke file dengan format yang rapi
fs.writeFileSync('./data.json', JSON.stringify(data, null, 2), 'utf-8');

console.log(`✅ Berhasil mengurutkan ulang ${data.length} soal dengan ID dari 1 sampai ${data.length}`);

File: reorder_ids.py
import json

# Baca file data.json
with open('data.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Urutkan ulang ID dari 1 sampai jumlah total soal
for index, question in enumerate(data, start=1):
    question['id'] = index

# Tulis kembali ke file dengan format yang rapi
with open('data.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

print(f"Berhasil mengurutkan ulang {len(data)} soal dengan ID dari 1 sampai {len(data)}")
